src/screen.js

```diff
@@ -0,0 +1,65 @@
+
+import config from './config'
+
+const AlpineScreenMagicMethod = {
+    start() {
+        let initialized
+
+        Alpine.addMagicProperty('screen', function ($el) {
+            // bind resize event to window with debounce
+            let update
+            const updateScreen = () => {
+                clearTimeout(update)
+                update = setTimeout(() => {
+                    $el.__x.updateElements($el)
+                }, 150)
+            }
+
+            // bind resize event to window if not initialized
+            if (!initialized) {
+                window.addEventListener('resize', updateScreen)
+            }
+
+            // set initialized to prevent multiple calls
+            initialized = true
+
+            return function (target) {
+                // Get current window dimensions
+                const width = window.innerWidth
+                const height = window.innerHeight
+
+                // If size provided as number, early return
+                if (Number.isInteger(target)) return target <= width
+
+                // Breakpoints and extras
+                const screen = {
+                    ...config.get('breakpoints'),
+                    touch: 'ontouchstart' in window,
+                    portrait: height > width,
+                    landscape: width > height,
```

Comment:
Wouldn't using `window.matchMedia("(orientation: portrait)").matches` be more of a native way to get this?

Reply:
Oh, you probably didn't see 

> > About the second question, How would you implement landscape or touch in your helper?
> 
> ```js
>                 const screen = {
>                     ...config.get('breakpoints'),
>                     touch: 'ontouchstart' in window,
>                     portrait: height > width,
>                     landscape: width > height
>                 }
> ```
> 
> i already finish the $screen helper, now working on tests.
> 
> I have also tried to use matchMedia for `landscape` and `portrait` to check but to test it i have to mock matchMedia method which didn't work properly.
> If i could make it work, i would try to use media queries as well for breakpoints
> 
> i am trying to achieve this:
> 
> ```js
>                 const screen = {
>                     ...config.get('breakpoints'),
>                     touch: 'ontouchstart' in window,
>                     portrait: window.matchMedia("(orientation: portrait)").matches,
>                     landscape: window.matchMedia("(orientation: landscape)").matches
>                 }
> ```
> 
> Works on browser but i couldn't make it work on jest test. Specially don't know how to test `touch`. Maybe i just push the code so far and see if we can improve it.

Oh, probably you haven't read this. We also wanted that but can't test it. If you know how to test it, we can change it tho.

Reply:
Ah ok. I read it at the time but forgot about it. I don't think we should change the implementation because we can't figure out how to test it though. I'll try and see what others do this week.

Reply:
We can test it but we have to mock match media (so a single test will cover landscape and portrait since there won't be any difference). We are mocking height and width anyway.
Out of curiosity, there is a pixel where the window is neither portrait nor landscape. It's worth investigating what mediaqueries do in this scenario and keep our implementation consistent if we go for a width/height comparison

Reply:
Not sure about that, but I know Chrome lets you override the orientation in DevTools, including custom orientations.

https://developers.google.com/web/tools/chrome-devtools/device-mode/orientation#:~:text=The%20Sensors%20tab%20opens%20up,provide%20your%20own%20exact%20orientation.

That said, I just tested it now and `window.matchMedia` didn't seem to be affected by the change.

In Chrome, 1000x1000 was portrait, but not sure about other devices. It does feel like something we should let the device decide though if we can. Is there any reason not to use window.matchMedia other than the test? 

We could add in Cypress and start testing too.

Reply:
Found a thing, working on it.

```js
                const screen = {
                    ...config.get('breakpoints'),
                    portrait: window.screen.orientation.angle === 90,
                    landscape: window.screen.orientation.angle === 0,
                }
```

so i can detect real orientation changes instead of comparing height and width. Mobile devices wont trigger resize event on screen orientation change, so better to do like so:
```js
        window.addEventListener('resize', debouncedScreensUpdate())
        window.addEventListener('orientationchange', debouncedScreensUpdate())
```

Reply:
I think we are green now. We can test screen orientation in proper way now. 🎉

Reply:
> Mobile devices wont trigger resize event on screen orientation change

Are you sure about this? We could probably add the listener anyway though.

`window.screen.orientation.angle` always gives `0` for me in Chrome

Another option is to just not include this part at all and instead let users add it themselves if they want it. We could just mention it in the docs.

Reply:
Try chrome mobile view and change orientation there. You will see 90 differences

Reply:
If Chrome always returns 0 when not in mobile mode, the code you're suggesting won't work properly. Both `portrait` and `landscape` could return true.

Personally I'm okay without this test and using `window.matchMedia("(orientation: portrait)").matches`, but if other's want it to be tested, and it's not feasible, then we should just let the user pass it in as an option. This seems like an edge case use anyway.

Reply:
Angle doesn't do what we want on desktop, it returns the orientation of the monitor (hardware) but the browser window could be resized to be in portrait mode, it would create confusion within users. 

If we use matchMedia there's really little to test, it's native so it does work. Mocking the object to test the condition passed in and return true and false, making sure the little code around preserve the right result would be enough for me. 

Breakpoints could use matchMedia too. We would need more discussion on this topic too so let's ignore it for now. 

I would say: let's get out a first version only supporting breakpoints which was the scope of the original PR. From there, let's open a discussion about orientation (same helper or different one? Listener we need, etc) and touch (possible $support helper). Would that work for you @muzafferdede?

Reply:
Sure thing. I think except orientation, we have a solid breakpoints which as you said original scope of this pr. should i clean up the code and just push a breakpoints only $screen helper now?

Reply:
I think breakpoints only when you have time please. Thanks for keeping up with all the changes.